src/Glpi/Asset/CustomField.php

```diff
@@ -0,0 +1,363 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Asset;
+
+use CommonDBChild;
+use CommonGLPI;
+use DirectoryIterator;
+use Dropdown;
+use Glpi\Application\View\TemplateRenderer;
+use Glpi\Asset\CustomField\TypeInterface;
+use Glpi\DBAL\QueryExpression;
+use Glpi\DBAL\QueryFunction;
+use ReflectionClass;
+use Session;
+
+final class CustomField extends CommonDBChild
+{
+    public static $itemtype = AssetDefinition::class;
+    public static $items_id = 'assets_assetdefinitions_id';
+
+    public static $rightname = 'config';
+
+    public static function getTypeName($nb = 0)
+    {
+        return _n('Custom field', 'Custom fields', $nb);
+    }
+
+    public static function getIcon()
+    {
+        return 'ti ti-forms';
+    }
+
+    public static function canCreate(): bool
+    {
+        return parent::canUpdate();
+    }
+
+    public static function canDelete(): bool
+    {
+        return parent::canUpdate();
+    }
+
+    public static function canPurge(): bool
+    {
+        return parent::canUpdate();
+    }
+
+    public function cleanDBonPurge()
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        $DB->update('glpi_assets_assets', [
+            'custom_fields' => QueryFunction::jsonRemove([
+                'custom_fields',
+                new QueryExpression($DB::quoteValue('$."' . $this->fields['id'] . '"'))
+            ])
+        ], [
+            'assets_assetdefinitions_id' => $this->fields['assets_assetdefinitions_id'],
+        ]);
+    }
+
+    public static function getAllowedDropdownItemtypes()
+    {
+        /**
+         * @var array $CFG_GLPI
+         */
+        global $CFG_GLPI;
+
+        static $allowed_dropdown_itemtypes = null;
```

Comment:
Please do not use `static` variables.

This could probably be moved into the `AssetDefinitionManager` constructor, as we do for the capacities. It would be easier to handle in the testing context.